docs/conf.py

```diff
@@ -208,6 +209,44 @@ def generate_doxygen_xml(app):
         raise Exception("An error has occurred while executing Doxygen\n" + str(e))
 
 
+def generate_r_docs(app):
+    """Generate documentation for R-package.
+
+    Parameters
+    ----------
+    app : object
+        The application object representing the Sphinx process.
+    """
+    commands = """
+    export PATH="/home/docs/.conda/bin:$PATH"
+    echo 'options(repos = "https://cran.rstudio.com")' > $HOME/.Rprofile 
+    conda create -q -y -n r_env r-base r-devtools r-data.table r-jsonlite r-magrittr r-matrix r-testthat cmake
```

Comment:
Without `testthat` documentation process crashes:
```
Found library file: /home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/lightgbm_r/src/lib_lightgbm.so to move to /tmp/RtmpgL11IH/devtools_install_4d06ff4063f/lightgbm/libs
Warning: unknown option ‘--keep_empty-dirs’

* installing to library ‘/home/docs/.conda/envs/r_env/lib/R/library’
* installing *source* package ‘lightgbm’ ...
** libs
installing via 'install.libs.R' to /home/docs/.conda/envs/r_env/lib/R/library/lightgbm
** R
** data
** demo
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (lightgbm)
Updating lightgbm documentation
Loading lightgbm
Re-compiling lightgbm
'/home/docs/.conda/envs/r_env/lib/R/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/home/docs/checkouts/readthedocs.org/user_builds/lightgbm/checkouts/docs/lightgbm_r'  \
  --library='/tmp/RtmpgL11IH/devtools_install_4d06ff4063f' --no-R --no-data  \
  --no-help --no-demo --no-inst --no-docs --no-exec --no-multiarch  \
  --no-test-load --preclean 

* installing *source* package ‘lightgbm’ ...
** libs
installing via 'install.libs.R' to /tmp/RtmpgL11IH/devtools_install_4d06ff4063f/lightgbm
* DONE (lightgbm)
Loading required package: R6
Error in loadNamespace(name) : there is no package called ‘testthat’
Calls: build_reference ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
Execution halted
```

Is it expected?

Reply:
@jameslamb The first one:
> "I had to add r-testthat to my install command and that worked but it's weird"

Everything is fine with `testthat`. You can observe it on the site right now.

I just wanted to know is there a workaround to drop this dependency. Thank for the link - now I know that's I'm not alone with this problem. 😃